tests/execute.rs

```diff
@@ -3,9 +3,73 @@ mod mock;
 
 use frame_support::assert_ok;
 use mock::*;
-use move_core_types::language_storage::TypeTag;
+use move_core_types::{
+    account_address::AccountAddress,
+    identifier::Identifier,
+    language_storage::{StructTag, TypeTag},
+};
 use move_vm_backend::types::MAX_GAS_AMOUNT;
 use pallet_move::transaction::Transaction;
+use sp_core::{crypto::Ss58Codec, sr25519::Public};
+use sp_runtime::AccountId32;
+
+fn addr32_from_ss58(ss58addr: &str) -> AccountId32 {
+    let (pk, _) = Public::from_ss58check_with_version(ss58addr).unwrap();
+    pk.into()
+}
+
+fn addr32_to_move(addr32: &AccountId32) -> Result<AccountAddress, pallet_move::Error<Test>> {
+    MoveModule::to_move_address(addr32)
+}
+
+fn addrs_from_ss58(ss58: &str) -> Result<(AccountId32, AccountAddress), pallet_move::Error<Test>> {
+    let addr_32 = addr32_from_ss58(ss58);
+    let addr_mv = addr32_to_move(&addr_32)?;
+    Ok((addr_32, addr_mv))
+}
+
+fn mv_script_name_to_transaction(
+    project: &str,
+    script: &str,
+    type_generic_args: Vec<TypeTag>,
+    param: Vec<u8>,
+) -> Result<Vec<u8>, anyhow::Error> {
+    let params: Vec<&[u8]> = vec![&param];
+    let script = assets::read_script_from_project(project, script);
+    let transaction = Transaction {
+        script_bc: script,
+        type_args: type_generic_args,
+        args: params.iter().map(|x| x.to_vec()).collect(),
+    };
+    Ok(bcs::to_bytes(&transaction)?)
+}
+
+fn get_vm_resource(
+    module_owner: &AccountAddress,
+    module_id: &str,
+    key_id: &str,
+    address: &AccountId32,
+) -> Result<Option<Vec<u8>>, anyhow::Error> {
+    let tag = StructTag {
+        address: *module_owner,
+        module: Identifier::new(module_id)?,
+        name: Identifier::new(key_id)?,
+        type_params: vec![],
+    };
+    let bytes = bcs::to_bytes(&tag)?;
+    let o_vec = MoveModule::get_resource(address, &bytes)
+        .map_err(|e| anyhow::anyhow!("MoveModule::get_resource {:?}", e))?;
+    Ok(o_vec)
+}
+
+fn script_transaction(
```

Comment:
A bit misleading name - it seems like a generic function for creating any script transaction, but it can used only for single parameter address/signer scripts

Reply:
Done